lib/pf/UnifiedApi/Controller/Fingerbank.pm

```diff
@@ -0,0 +1,281 @@
+package pf::UnifiedApi::Controller::Fingerbank;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Fingerbank -
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Fingerbank
+
+=cut
+
+use strict;
+use warnings;
+use pf::error qw(is_error);
+use pf::util qw(expand_csv);
+use Mojo::Base 'pf::UnifiedApi::Controller::RestRoute';
+use Mojo::Util qw(url_unescape);
+use pf::UnifiedApi::Search::Builder::Fingerbank;
+
+=head2 url_param_name
+
+The name of the url parameter to get the value of the resource identifier
+
+=cut
+
+has 'url_param_name';
+
+=head2 primary_key
+
+The name of the primary key in the data base
+
+=cut
+
+has 'primary_key';
+
+
+=head2 fingerbank_model
+
+fingerbank_model
+
+=cut
+
+has 'fingerbank_model';
+
+=head2 search_builder_class
+
+search_builder_class
+
+=cut
+
+has 'search_builder_class' => "pf::UnifiedApi::Search::Builder::Fingerbank";
+
+
+sub resource {
+    my ($self) = @_;
+    my $id = $self->stash($self->url_param_name);
+    my $model = $self->fingerbank_model;
+    my ($status, $item ) = $model->read_hashref($id);
+    if (is_error($status)) {
+        return $self->render_error($status, $item);
+    }
+
+    push @{$self->stash->{item}}, $item;
+    return 1;
+}
+
+=head2 list
+
+list
+
+=cut
+
+sub list {
+    my ($self) = @_;
+    my ($status, $search_info_or_error) = $self->build_list_search_info;
+    if (is_error($status)) {
+        return $self->render(json => $search_info_or_error, status => $status);
+    }
+
+    ($status, my $response) = $self->search_builder->search($search_info_or_error);
+    if ( is_error($status) ) {
+        return $self->render_error(
+            $status,
+            $response->{msg},
+            $response->{errors}
+        );
+    }
+
+    return $self->render(json => $response);
+}
+
+=head2 create
+
+create
+
+=cut
+
+sub create {
+    my ($self) = @_;
+    my ($status, $json) = $self->parse_json;
+    if (is_error($status)) {
+        return $self->render(json => $json, status => $status);
+    }
+
+    ($status, my $return) = $self->fingerbank_model->create($json);
+    if (is_error($status)) {
+        return $self->render_error($status, $return);
+    }
+
+    my $parent_route = $self->match->endpoint->parent->name;
+    my $url = $self->url_for("$parent_route.resource.get", {$self->url_param_name => $return->{id}});
+    $self->res->headers->location($url);
+    return $self->render(json => {});
+}
+
+=head2 remove
+
+remove
+
+=cut
+
+sub remove {
+    my ($self) = @_;
+    my $id = $self->id;
+    my ($status, $msg) = $self->fingerbank_model->delete($id);
+    if (is_error($status)) {
+        return $self->render_error($status, $msg);
+    }
+
+    return $self->render(json => {});;
```

Comment:
Not sure its a plain 200 the other deletion calls return

Reply:
It is

Reply:
It is